scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,214 @@
+    function DownloadRepository([string]$repository, [string]$commitHash, [string]$sourceDirectoryPath)
+    {
+        if(!(Test-Path $sourceDirectoryPath))
+        {
+                git clone $repository $sourceDirectoryPath
+                git -c $sourceDirectoryPath checkout $commitHash
+        }
+        else 
+        {
+                Log "Skipping the cloning of $repository as $sourceDirectoryPath is not empty" -color "Yellow"
+        }
+    }
+    
+    # Find the appropriate solution file for the repository. Looks for a solution file matching the repo name, 
+    # if not it takes the first available sln file in the repo. 
+    function GetSolutionFile([string]$repository,[string]$sourceDirectoryPath) {
+
+        $gitRepoName = $repository.Substring($($repository.LastIndexOf('/') + 1))
+        $potentialSolutionFile = [System.IO.Path]::Combine($sourceDirectoryPath, "$($gitRepoName.Substring(0, $gitRepoName.Length - 4)).sln")
+
+        if(Test-Path $potentialSolutionFile)
+        {
+            $solutionFile = $potentialSolutionFile
+        } 
+        else 
+        {
+            $possibleSln = Get-ChildItem $sourceDirectoryPath *.sln
+            if($possibleSln.Length -eq 0)
+            {
+                Log "No solution files found in $sourceDirectoryPath" "red"
+            } 
+            else 
+            {
+            $solutionFile = $possibleSln[0] | Select-Object -f 1 | Select-Object -ExpandProperty FullName
+            }
+        }
+        return $solutionFile;
+    }
+
+    # Given a repository and a hash, checks out the revision in the given source directory. The return is a solution file if found. 
+    function SetupGitRepository([string]$repository, [string]$commitHash, [string]$sourceDirectoryPath)
+    {
+        Log "Setting up $repository into $sourceDirectoryPath"
+        DownloadRepository $repository $commitHash $sourceDirectoryPath
+        $solutionFile = GetSolutionFile $repository $sourceDirectoryPath
+        Log "Completed the repository setup. The solution file is $solutionFile" -color "Green"
+        return $solutionFile
+    }
+
+    # runs locals clear all with the given client
+    function LocalsClearAll([string]$nugetClient)
+    {
+        $nugetClient = GetAbsolutePath $nugetClient
+        if($(IsClientDotnetExe $nugetClient))
+        {
+            . $nugetClient nuget locals -c all *>>$null
+        } 
+        else 
+        {
+            . $nugetClient locals -clear all -Verbosity quiet
+        }
+    }
+
+    # Gets the client version
+    function GetClientVersion($nugetClient)
+    {
+        $nugetClient = GetAbsolutePath $nugetClient
+        if(IsClientDotnetExe $nugetClient)
+        {
+            $version = . $nugetClient --version
+            Log "$version"
+            return $version
+        } 
+        else 
+        {
+            $versionQuery = . $nugetClient
+            $version = $(($versionQuery -split '\n')[0]).Substring(15)
```

Comment:
.Substring("NuGet Version: ".Length)

That shows where the mysterious `15` comes from.